src/main/java/org/craftercms/studio/controller/rest/v2/GroupsController.java

```diff
@@ -72,18 +75,27 @@ public ResponseBody getAllGroups(
             @RequestParam(value = "limit", required = false, defaultValue = "10") int limit,
             @RequestParam(value = "sort", required = false, defaultValue = StringUtils.EMPTY) String sort
             ) throws ServiceLayerException {
-        int total = groupService.getAllGroupsTotal(DEFAULT_ORGANIZATION_ID);
-        List<Group> groups = groupService.getAllGroups(DEFAULT_ORGANIZATION_ID, offset, limit, sort);
+        int total = 0;
+        try {
+            total = groupService.getAllGroupsTotal(DEFAULT_ORGANIZATION_ID);
+            List<GroupTO> groups = groupService.getAllGroups(DEFAULT_ORGANIZATION_ID, offset, limit, sort);
 
-        ResponseBody responseBody = new ResponseBody();
-        PaginatedResultList<Group> result = new PaginatedResultList<>();
-        result.setTotal(total);
-        result.setOffset(offset);
-        result.setLimit(CollectionUtils.isEmpty(groups) ? 0 : groups.size());
-        result.setResponse(ApiResponse.OK);
-        result.setEntities(groups);
-        responseBody.setResult(result);
-        return responseBody;
+            ResponseBody responseBody = new ResponseBody();
+            PaginatedResultList<GroupTO> result = new PaginatedResultList<>();
+            result.setTotal(total);
+            result.setOffset(offset);
+            result.setLimit(CollectionUtils.isEmpty(groups) ? 0 : groups.size());
+            result.setResponse(ApiResponse.OK);
+            result.setEntities(groups);
+            responseBody.setResult(result);
+            return responseBody;
+        } catch (OrganizationNotFoundException e) {
```

Comment:
Can't we move this to ExceptionHandlers? That's where we're currently putting all our API exception handling code.